### PR TITLE
N Makefile

### PR DESCRIPTION
**Pull request by matthew-andrews**

### CODE REVIEW

File: n.Makefile [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	@git clean -fxd
+	@echo $(DONE)
+
+# install
+instal%: node_modules bower_components _install_scss_lint
+	@for n in $(shell ls functions 2>/dev/null); do $(MAKE) functions/$$n/node_modules; done
+	@echo $(DONE)
+
+# deploy
+deplo%:
```
Comment (adgad) [inline n.Makefile]: Out of interest, what is the % here?

**Comment by adgad** [comment]
:+1:

File: n.Makefile (hunk 1 above)
Comment (matthew-andrews) [inline n.Makefile]: for other's benefit, it's to help with inheritance and overriding of tasks

File: n.Makefile [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#
+# Two main types of tasks.  Common and sub.
+#
+# Common tasks are the public facing API of the Makefile.  Developers/CI should run these commands.
+# Sub-tasks are the inner, private workings of n-makefile.  Developers/CI ought not to run these commands.
+#
+# Type specific conventions
+#
+# - For ‘common’ tasks
+#     - name must match the directory name they generate
```
Comment (ironsidevsquincy) [inline n.Makefile]: does this mean `make build` should create a `build` dir, rather than `public`?

File: n.Makefile [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#
+
+# clean
+clea%:
+	@git clean -fxd
+	@echo $(DONE)
+
+# install
+instal%: node_modules bower_components _install_scss_lint
+	@for n in $(shell ls functions 2>/dev/null); do $(MAKE) functions/$$n/node_modules; done
```
Comment (ironsidevsquincy) [inline n.Makefile]: what does this line do?

File: n.Makefile (hunk 2 above)
Comment (matthew-andrews) [inline n.Makefile]: ah no, this is for sub-tasks not common ones:-
https://github.com/Financial-Times/n-makefile/blob/master/Makefile#L21-L22

i think maybe make build could be a special case but we could do something like:-

```
build:
	@for n in $(shell ls client/[^_]*.scss 2>/dev/null); do $(MAKE) public/$$n; done
	@for n in $(shell ls client/*.js 2>/dev/null); do $(MAKE) public/$$n; done

public/%.scss:
	obt build --scssFile $@ blah blah blah

public/%.js:
	obt build --jsFile $@ blah blah blah
```

**Comment by ironsidevsquincy** [comment]
:+1:

**Pull request merged**